### ISSUE TITLE
Flag explicit usage of naming parameter as it in a lambda function

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#954]
Explicitly declaring a lambda function parameter name as `it` should be flagged since it's the default and not necessary to have it.

e.g.

```kotlin
return foo.flatMapObservable { it -> Observable.fromIterable(it.bar) }
```
